codec-dns/src/main/java/io/netty/handler/codec/dns/TcpDnsQueryDecoder.java

```diff
@@ -0,0 +1,81 @@
+package io.netty.handler.codec.dns;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.CorruptedFrameException;
+import io.netty.handler.codec.LengthFieldBasedFrameDecoder;
+
+public final class TcpDnsQueryDecoder extends LengthFieldBasedFrameDecoder {
+    private final DnsRecordDecoder decoder;
+
+    public TcpDnsQueryDecoder() {
+        this(DnsRecordDecoder.DEFAULT, 64 * 1024);
+    }
+
+    public TcpDnsQueryDecoder(DnsRecordDecoder decoder, int maxFrameLength) {
+        super(maxFrameLength, 0, 2, 0, 2);
+        this.decoder = decoder;
```

Comment:
```suggestion
        this.decoder = ObjectUtil.requireNonNull(decoder, "decoder");
```

Reply:
I can not found ObjectUtil.requireNonNull,  I use ObjectUtil.checkNotNull instead of it.